internal/backends/postgresql/metadata/metadata.go

```diff
@@ -60,5 +72,109 @@ func (c *Collection) Unmarshal(doc *types.Document) error {
 	c.Name = must.NotFail(doc.Get("_id")).(string)
 	c.TableName = must.NotFail(doc.Get("table")).(string)
 
+	if doc.Has("settings") {
+		var settings Settings
+		must.NoError(settings.Unmarshal(must.NotFail(doc.Get("settings")).(*types.Document)))
+		c.Settings = settings
+	} else {
+		// If settings are not present, we initialize them with empty indexes to avoid potential nil pointers.
+		c.Settings = Settings{Indexes: []IndexInfo{}}
+	}
+
 	return nil
 }
+
+// Settings represents collection settings.
+type Settings struct {
+	Indexes []IndexInfo `json:"indexes"`
+}
+
+// deepCopy returns a deep copy.
+func (s Settings) deepCopy() Settings {
+	indexes := make([]IndexInfo, len(s.Indexes))
+
+	for i, index := range s.Indexes {
+		indexes[i] = IndexInfo{
+			Name:   index.Name,
+			Key:    slices.Clone(index.Key),
+			Unique: index.Unique,
+		}
+	}
+
+	return Settings{
+		Indexes: indexes,
+	}
+}
+
+// Marshal returns [*types.Document] for settings.
+func (s Settings) Marshal() *types.Document {
+	indexes := types.MakeArray(len(s.Indexes))
+
+	for _, index := range s.Indexes {
+		key := types.MakeArray(len(index.Key))
+
+		for _, pair := range index.Key {
+			key.Append(must.NotFail(types.NewDocument(
+				"field", pair.Field,
+				"descending", pair.Descending,
+			)))
+		}
+
+		indexes.Append(must.NotFail(types.NewDocument(
+			"name", index.Name,
+			"dbindex", index.DBIndex,
+			"key", key,
+			"unique", index.Unique,
+		)))
+	}
+
+	return must.NotFail(types.NewDocument(
+		"indexes", indexes,
+	))
+}
+
+// Unmarshal sets settings from [*types.Document].
+func (s *Settings) Unmarshal(doc *types.Document) error {
+	indexes := must.NotFail(doc.Get("indexes")).(*types.Array)
+
+	s.Indexes = make([]IndexInfo, indexes.Len())
+
+	iter := indexes.Iterator()
+	defer iter.Close()
+
+	for {
+		i, v, err := iter.Next()
+		if errors.Is(err, iterator.ErrIteratorDone) {
+			break
+		}
+
+		if err != nil {
+			return lazyerrors.Error(err)
+		}
+
+		doc := v.(*types.Document)
+
+		s.Indexes[i] = IndexInfo{
+			Name:    must.NotFail(doc.Get("name")).(string),
+			DBIndex: must.NotFail(doc.Get("dbindex")).(string),
+			Key:     must.NotFail(doc.Get("key")).([]IndexKeyPair),
+			Unique:  must.NotFail(doc.Get("unique")).(bool),
+		}
+	}
+
+	return nil
+}
+
+// IndexInfo represents information about a single index.
+type IndexInfo struct {
+	Name    string         `json:"name"`
+	DBIndex string         `json:"dbindex"` // how the index is created in the DB, like TableName for Collection
```

Comment:
I need some help with naming here :) (cc @chilagrow @noisersup @AlekSi).
The idea of this field is the same as `TableName` in the collection structure. It's a hashed name of the index for the PG database (`CREATE INDEX <DBIndex> ...`). But I'm not sure how to call it correctly. `Name` is the name in terms of MongoDB (it could be very long) and `DBIndex` is an index name possible in PG.

Reply:
`PgIndexName`, `IndexName`,`TableIndexName` hmm. `CollectionIndexName` vs `TableIndexName` might differentiate better but very long.

Either way, let's leave a comment what they are. I was not sure what previous `pgIndex` was, until your explanation now :see_no_evil:

Reply:
Yeah, and a part of the problem is `pgindex` as a field stored in JSON in the DB [must remain the same](https://ferretdb.slack.com/archives/C02NWN1K3MJ/p1695810442921169).

Nice suggestion for the field name! I especially like `TableIndexName` because it gives a clear link to `TableName` :) 


Collection: Name -> TableName (stored as `table` in the DB's json)
Index: Name -> TableIndexName (stored as `pgindex` in the DB's json)